athletics.lic

```diff
@@ -0,0 +1,223 @@
+        climb_practice(5287, 'rock gorge',true)
+      else
+	    if DRSkill.getrank('Stealth') >= 130
+	      climb_practice(5143, 'deep crack',true)
+		end
+      end
+    end
+  end
+  
+  def climb_practice(room, target, to_hide = false)
+    return unless target
+    walk_to(room)
+    play_song?
+    while DRSkill.getxp('Athletics') <= @end_skill
+      retreat
+      if to_hide
+        attempts_to_hide = 5
+        while attempts_to_hide > 0
+          break if hide?
+          attempts_to_hide -= 1
+        end
+        return if attempts_to_hide.zero?
+      end
+      Flags.add('ct-climbing-finished', 'You finish practicing your climbing')
+      Flags.add('ct-climbing-combat', 'You are engaged in combat')
+      bput("climb practice #{target}", 'You begin to practice ')
+      loop do
+        pause 1
+        break if Flags['ct-climbing-finished']
+        return if Flags['ct-climbing-combat']
+        if DRSkill.getxp('Athletics') > @end_skill
+          fput('stop climb')
+          break
+        end
+      end
+    end
+    stop_play
+    bput('unhide', 'You come out of hiding', 'You slip out of hiding', 'But you are not')
+    walk_to(@settings.safe_room)
+  end
+  
+  def climb?(room, targets)
+    targets.each do |target|
+      walk_to(room)
+      return true if DRRoom.npcs.length >= 3
+
+      fput "climb #{target}"
+      pause
+      waitrt?
+      return false if DRSkill.getxp('Athletics') >= @end_skill
+    end
+    true
+  end
+  
+  def play_song?(blocking = false)
+    UserVars.song = @song_list.first.first unless UserVars.song
+    @did_play = true
+    case bput("play #{UserVars.song}", 'dirtiness may affect your performance', 'slightest hint of difficulty', 'You begin a', 'You struggle to begin', 'You\'re already playing a song', 'You effortlessly begin', 'You begin some', 'You cannot play', 'Play on what instrument', 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play')
+    when 'Play on what instrument'
+      @no_instrument = true
+      return false
+    when 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play'
+      return true
+    when 'dirtiness may affect your performance'
+      if DRSkill.getrank('Performance') < 20
+        echo "Skipping cleaning of zills due to low rank of Performance: #{DRSkill.getrank('Performance')}"
+        return true
+      end
+      stop_play
+      clean_zills
+      return play_song?
+    when 'You begin a', 'You effortlessly begin', 'You begin some'
+      stop_play
+      UserVars.song = @song_list[UserVars.song] || @song_list.first.first
+      return play_song?
+    when 'You struggle to begin'
+      if UserVars.song != @song_list.first.first
+        stop_play
+        UserVars.song = @song_list.first.first
+        return play_song?
+      end
+    end
+ 
+    return true unless blocking
+ 
+    Flags.reset('song')
+    pause 1 until Flags['song']
```

Comment:
Probably copied from crossing-trainer, but we can swap all the `pause 1` with `pause`